components/registry-facade/pkg/registry/layersource.go

```diff
@@ -489,3 +559,85 @@ func (src *ContentLayerSource) GetBlob(ctx context.Context, spec *api.ImageSpec,
 	err = errdefs.ErrNotFound
 	return
 }
+
+// ParsedEnvs is parsed image envs configuration
+type ParsedEnvs struct {
```

Comment:
nit: it's a bit odd that `ParsedEnv` has no exposed fields or functions. I'd make `set`, `append`, and `prepent` public s.t. others can implement their own `EnvModifier`